rust/arrow/src/array/builder.rs

```diff
@@ -577,14 +579,15 @@ impl<T: ArrowPrimitiveType> PrimitiveBuilder<T> {
 
     /// Appends a value of type `T` into the builder
     pub fn append_value(&mut self, v: T::Native) -> Result<()> {
-        self.bitmap_builder.append(true);
+        self.bitmap_builder.as_mut().map(|b| b.append(true));
         self.values_builder.append(v);
         Ok(())
     }
 
     /// Appends a null slot into the builder
     pub fn append_null(&mut self) -> Result<()> {
-        self.bitmap_builder.append(false);
+        self.materialize_bitmap_builder();
```

Comment:
Could you also benchmark the case where there are nulls in the input? Wouldn't that be a bit slower now?

Reply:
The change is within the noise threshold. That is exactly what I expected, there is enough code around the buidlers that adding an extra branch for conditionally creating the array does not make much difference.

The benchmark results are:
```
Before:
bench_primitive_nulls/bench_primitive_nulls
                        time:   [4.5728 ms 4.5761 ms 4.5795 ms]

After:
bench_primitive_nulls/bench_primitive_nulls
                        time:   [4.5972 ms 4.6004 ms 4.6038 ms]
                        change: [+0.4255% +0.5290% +0.6284%] (p = 0.00 < 0.05)
                        Change within noise threshold.
```

Reply:
Thanks, makes sense!